src/sagas/utils.js

```diff
@@ -165,9 +167,38 @@ export function* fetchPermits ({ entityType, id }) {
 }
 
 export const PermissionsType = {
-  STORAGE_DOMAIN_TYPE: 'StorageDomain',
-  CLUSTER_TYPE: 'Cluster',
-  VNIC_PROFILE_TYPE: 'VnicProfile',
   DISK_TYPE: 'Disk',
-  TEMPLATE_TYPE: 'Template',
+}
+
+/**
+ * Convert a set of permissions to a set of permits for the app's current user by
+ * mapping the permissions through their assigned roles to the permits.
+ *
+ * NOTE: If the user is an admin user the user's group and id membership must be
+ * explicitly checked.
+ */
+export function* permissionsToUserPermits (permissions) {
+  const userFilter = yield select(state => state.config.get('filter'))
+  const userGroups = yield select(state => state.config.get('userGroups'))
+  const userId = yield select(state => state.config.getIn(['user', 'id']))
+  const roles = yield select(state => state.roles)
+
+  const permitNames = []
+  for (const permission of (Array.isArray(permissions) ? permissions : [permissions])) {
```

Comment:
I think it would be nice to move `(Array.isArray(permissions) ? permissions : [permissions])` to separate constant